Initial work to get 64 bit version to work

ilzu opened the pull request:
Initial work to get Sequitur to work on 64 bit Haiku. Now compiles and seems to work on x86 (both gcc2 and gcc4+) and compiles and runs (but doesn't show window) x86_64.

ilzu committed:
Quick and dirty port to 64 bit. Doesn't work yet.

ilzu committed:
Fixed compilation on 32 bit with gcc 4+

ilzu committed:
removed that one symbol that breaks patch on windows

ilzu committed:
Fixed some compilation errors on gcc2, fixed makefile and buildsystem a bit

ilzu committed:
Fixed some compilation errors on gcc2, fixed makefile and buildsystem a bit

ilzu committed:
Merge branch 'master' of github.com:ilzu/Sequitur

ilzu committed:
Merge branch 'master' of github.com:ilzu/Sequitur

ilzu committed:
Fix compilation error on 64 bit

ilzu committed:
Merge branch 'master' of github.com:ilzu/Sequitur

ilzu committed:
Fix compile on 64 bit

ilzu committed:
Merge branch 'master' of github.com:ilzu/Sequitur

ilzu committed:
Fix typo

ilzu committed:
now builds (and seems to work) on gcc2. Need to figure out why it's crashing on newer gccs.

ilzu committed:
Use int32 instead of long and uint32 instead of ulong, so less chances for int size mismatches on 64 bit

ilzu committed:
Merge branch 'master' of github.com:ilzu/Sequitur

ilzu committed:
Made sure Gehnaphores work on x86_64

ilzu committed:
Made sure Gehnaphores work on x86_64

ilzu committed:
Merge branch 'master' of github.com:ilzu/Sequitur

ilzu committed:
Merge branch 'master' of github.com:ilzu/Sequitur

ilzu committed:
Merge branch 'master' of github.com:ilzu/Sequitur

waddlesplash commented on AmFilters/ArpChorus.cpp:
```diff
@@ -289,10 +289,10 @@ status_t ArpChorusFilter::Configure(ArpVectorI<BView*>& panels)
-	The chorused notes can be offset in pitch by octave and/or step amounts,
-	and their velocities can be scaled. Typically, the velocity should be increased
-	for notes lower in pitch and decreased for notes higher in pitch.</P>";
+	str << "<P>I produce up to four additional notes for every note I receive. "
```
New block has no linebreaks nor even whitespace between words. That should be fixed.

ilzu replied:
Should be fixed now.

waddlesplash commented on AmFilters/ArpConsole.cpp:
```diff
@@ -400,10 +400,10 @@ status_t ArpConsoleFilter::Configure(ArpVectorI<BView*>& panels)
-	for debugging any filters you've written, so you can see the output
-	they are providing.  In order for this filter to work, Sequitur must be
-	launched from a command line.</P>";
+	str << "<P>I print in the terminal every MIDI event I receive. I am useful"
```
same

waddlesplash commented on AmFilters/ArpControllerEnvelope.cpp:
```diff
@@ -285,35 +285,35 @@ status_t ArpControllerEnvelopeFilter::Configure(ArpVectorI<BView*>& panels)
-velocity -- effectively, I transform a note into a controller sweep. One such application would
-be to write a single drum part, and 'carve' that part out of a long, evolving sound. Here is how
-you would achieve that effect:</P>
+	str << "<P><I>This filter is now obsolete.  Use Motion Control, instead.</I></P>"
```
same, and in following files.

ilzu replied:
And here too. And all AmFilters.

waddlesplash commented on ArpHeader/AmKernel/AmGlobalsImpl.h:
```diff
@@ -40,8 +40,8 @@
-typedef vector<AmViewFactory*>			fac_vec;
-typedef vector<BMessenger>				observer_vec;
+typedef std::vector<AmViewFactory*>			fac_vec;
+typedef std::vector<BMessenger>				observer_vec;
```
You can do `using std::vector` to avoid this, I think.

ilzu replied:
Yup, I know. But it was faster to just search and replace all instances of "vector<" with "std::vector<" and same for all other things in std namespace.

waddlesplash commented on ArpHeader/AmKernel/locks.h:
```diff
@@ -0,0 +1,172 @@
+/*
```
You need to use the private header from the Haiku headers directory instead of adding it to this repository.

ilzu replied:
The reason I borrowed the locks.h from Haiku repository is that as I borrowed also MultiLocker.cpp and MultiLocker.h (the MultiLocker class in Haiku is derived from the same Be code that is in  Sequitur repo but updated a lot.) It was easier to borrow the updated class from haiku than to fix the old one to work on 64 bits. I included the locks.h in here as if it changes in Haiku and is not compatible anymore, that wouldn't break the MultiLocker class. But I can remove the header from here and use the system one.

waddlesplash replied:
You have to use the system one because it is using kernel and libroot functions, and so if it breaks MultiLocker, it will need to be adjusted.

ilzu committed:
Enabled debug on MultiLocker.cpp and suddenly 64 bit version works.

ilzu committed:
Added space and newline to the ends of the textblocks in AmFilter::LongDescription functions

ilzu committed:
Use HAiku private header locks.h instead of having a copy of it here.

waddlesplash commented on ArpHeader/ArpBuild.h:
```diff
@@ -52,6 +52,8 @@
 
 #if __INTEL__
 
+#if __CNUC__ < 3
```
?

ilzu replied:
This is to use __declspec(dllimport) / __declspec(dllexport) only on gcc2.

pulkomandy replied:
There is a typo, it should be GNUC and not CNUC.

Also, these declspecs are useless on gcc2 and can be removed completely. I think they were used in some older version for BeOS (when it was not using ELF format for executables), but they are not used by the current version as far as I know.

ilzu replied:
Fixed the typo

waddlesplash commented:
This diff would be much smaller and more pleasant to read if you did the `using std::...` in a base header and then reverted back to the old invocations; so please do that.

korli commented:
> This diff would be much smaller and more pleasant to read if you did the `using std::...` in a base header and then reverted back to the old invocations; so please do that.

+1  Also using stdio.h vs cstdio shouldn't make a difference IMO.

pulkomandy commented:
Not sure I agree, putting "using namespace std" globally often ends up creating name conflicts with other functions and sometimes hard to debug ones (I had such problems in WebKit and it took a while to figure it out because it resulted in a compilation error in some C++ library include for redefining a function).

Using the std:: prefix is the right thing to do, even if it's a bit verbose.

waddlesplash commented:
> Not sure I agree, putting "using namespace std" globally 

Indeed we don't want to do that. Instead we want to do `using std::vector`, `using std::endl`, and import only what is needed, not the whole namespace.

ilzu commented on ArpHeader/AmKernel/AutoLocker.h:
```diff
@@ -0,0 +1,194 @@
+ * All rights reserved. Distributed under the terms of the MIT License.
+ */
+#ifndef _AUTO_LOCKER_H
+#define _AUTO_LOCKER_H
```
I must have overlooked it when I removed locks.h. Will fix in next commit.

ilzu commented:
I agree with Pulkomandy on this issue, I personally prefer using std::prefix in the code and dislike using using std::something. I think it makse the code less readable and might cause problems such as Pulkomandy mentioned. But I agree that it makes the patch much bigger and verbose.  And as Korli mentioned, those <cstdio> vs <stdio.h> do not make any difference, but as I tried to find where there are sysincludes that have been renamed (iostream.h, vector.h etc) I changed those too, just a personal preference .

waddlesplash commented:
It is better to keep changes separate in case one breaks something and they have to be edited later. So, even if we ultimately switch to the "verbose" usages, we should avoid doing that in this commit to keep it as small as possible.

ilzu committed:
Removed AutoLocker.h header an use Haiku header

smccrearyx commented:
I would assume that we'd want to bump to revision from v2.1.3 to v2.2.0 if/when this is accepted?  Should that be done in this pull request or a follow up one?

pulkomandy commented on ArpSrc/AmKernel/AmStdFilters.cpp:
```diff
@@ -913,24 +914,24 @@ BString AmProducerFilterAddOn::ShortDescription() const
-		<p>There are a variety of ways to add MIDI ports to your system:  Some soundcards
-		have supported MIDI ports, or drivers have been published for several popular USB-based
-		MIDI interfaces (such as the Yamaha UX256).</p>";
+	str << "<p>I correspond to a single BeOS MIDI producer.  The most common type of MIDI"
```
this one has no \n like hte others? At least it should have spaces at the end of lines or at the start of the next line, otherwise words will be fused together.

ilzu replied:
Fixed.

pulkomandy commented on ArpSrc/Sequitur/SeqFilterAddOnWindow.cpp:
```diff
@@ -318,8 +318,10 @@ SeqFilterAddOnWindow::SeqFilterAddOnWindow(	BRect frame,
 	mRoster = roster;
-	AmMultiFilterRoster*	roster = AmMultiFilterRoster::Default();
-	if (roster) roster->AddObserver(this);
+	//HACK to make compile on 64 bit. What the hell s going on here?
```
It looks like there were a local variable and a parameter both named roster and one needs to be renamed?
This is (was?) allowed by the standard but the compiler generates a warning now.

ilzu replied:
Removed the comment and commented out line, as it seems to be the right fix for it.

pulkomandy commented on Sequitur/SequiturFilters_CP_040320/ArpDrone.cpp:
```diff
@@ -132,11 +132,11 @@ ArpDroneAddOn::ArpDroneAddOn(const void* cookie)
-<P>Example use: place Drone in a bass track, send its output to a drum track, 
-select a note which plays a bass drum in Drones GUI - the bass drum with be
-automatically snychronized with the bass.</P>";
+	str << "<P>Send copies of all note events to another pipeline, forcing all notes"
```
missing \n or spaces here as well

ilzu replied:
Missed this one, but fixed now.

pulkomandy commented on Sequitur/SequiturFilters_CP_040320/ArpVelociTease.cpp:
```diff
@@ -167,19 +167,19 @@ void ArpVelociTeaseFilter::Start(uint32 context)
-				have no effect.</LI>
-		<LI><I>Random</I> will assign a random velocity to each note.</LI>
-	</UL>&nbsp;";
+	str << "<P>I alter the velocity of all incoming notes.  The change I make depends on"
```
missing \n or spaces

ilzu replied:
Fixed.

pulkomandy commented on makefiles/platform:
```diff
@@ -185,8 +187,89 @@ endif
+
+#		specify the optimizer setting
+ifdef RELEASE_BUILD
+#		CPLATFLAGS	+= -O3 -DNDEBUG=1 -- O3 crashes compiler! NDEBUG not in Sequitur
```
NDEBUG disables assert() from <assert.h>, if that's used

ilzu replied:
I know it does. That is an old left over comment, I just copied from the x86 block for x86_64 block and forgot to remove. Probably from back from BeOS days, as I don't remember hearing -O3 crashing compiler since.  Removed the comment.

pulkomandy commented:
The file "Sequitur/Skins/Plate Metal" is added, is this intentional?

I made a few comments inline as well. Mainly some documentation strings are missing some spaces or newlines. Other than that, the change seems good to go to me. Sorry for the delay in reviewing, it is a large change and I could not set aside enough time for it until now.

Did anyone contact Pete Goodeve about this? He did most of the work on getting Sequitur back to properly running again (https://www.goodeveca.net/Sequitur/) so maybe he wants to check this before we do a release? Not sure if he has a github account however.

ilzu committed:
Fixes for PulkoMandy's comments
added line endings and spaces to generated html documentation in
	ArpSrc/AmKernel/AmStdFilters.cpp and
	Sequitur/SequiturFilters_CP_040320/ArpVelociTease.cpp
fixed typo on ArpHeaders/ArpBuild.h and removed unneccessary comment and
commented out line in ArpSrc/Sequitur/SeqFilterAddOnWindow.cpp

ilzu committed:
Removed ancient comment

ilzu commented:
> The file "Sequitur/Skins/Plate Metal" is added, is this intentional?
I don't recall touching that file. And it seems to be a copy of example skin and not the plate metal skin. So not intentional.

ilzu committed:
restored Sequitur/Skins/Plate Metal

ilzu committed:
newlines and spaces to generated htmldocs

pulkomandy commented:
The change is already cleanly split in smaller commits if you want to review it that way.
